### PR TITLE
Add version and dry-run.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
Improved version of https://github.com/ambiata/ambiata-cli/pull/23

### CODE REVIEW

File: main/ambiata-daemon.hs [hunk 1]
```diff
@@ -32,9 +38,15 @@ main = do
             pure $ setFormatter h logFmt
   updateGlobalLogger rootLoggerName (setHandlers [logh])
 
-  execParser (info (helper <*> cmd) (fullDesc <> mainDesc <> header "ambiata-daemon - Ambiata CLI Daemon")) >>= \case
+  dispatch cli >>= \case
+    VersionCommand -> putStrLn ("ambiata-cli: " <> buildInfoVersion) >> exitSuccess
```
Comment (charleso) [inline main/ambiata-daemon.hs]: This is `ambiata-daemon`

File: main/ambiata-daemon.hs [hunk 2]
```diff
@@ -32,9 +38,15 @@ main = do
...
+    RunCommand DryRun c -> print c >> exitSuccess
```
Comment (charleso) [inline main/ambiata-daemon.hs]: Not sure if this really makes sense for daemon, but I don't have any suggestions.

**Comment by charleso** [comment]
:+1:

**Comment by HuwCampbell** [comment]
If you're doing things on the parser, you can also put both commands inside the one subparser now that the `internal` is gone.

**Comment by markhibberd** [comment]
@tmcgilchrist Want to rebase and get this in?

**Pull request merged**